src/sentry/api/endpoints/project_trace_item_details.py

```diff
@@ -15,15 +17,35 @@
 from sentry.api.bases.project import ProjectEndpoint
 from sentry.api.exceptions import BadRequest
 from sentry.models.project import Project
+from sentry.search.eap.types import SupportedTraceItemType
+from sentry.search.eap.utils import translate_internal_to_public_alias
 from sentry.snuba.referrer import Referrer
 from sentry.utils import snuba_rpc
 
 
-def convert_rpc_attribute_to_json(source: dict) -> dict:
-    for k, v in source.items():
-        if k.startswith("val"):
-            return {"type": k[3:].lower(), "value": v}
-    raise BadRequest(f"unknown field in protobuf: {source}")
+def convert_rpc_attribute_to_json(
+    attributes: list[dict],
+) -> Generator[dict]:
+    for attribute in attributes:
+        source = attribute["value"]
+        for k, v in source.items():
+            if k.startswith("val"):
+                val_type = k[3:].lower()
```

Comment:
Should the. `.lower()` be lifted? This condition means that only lower cased values will make it through.

Reply:
I'm not sure I follow, that would leave us with ambiguously cased types? This `val_type` is str compared, so we have to lower it, and the resulting value in the payload shouldn't retain casing?